File: fund-frontier/crowdfund/src/main/java/com/project/crowdfund/service/FundsService.java
package com.project.crowdfund.service;

import java.util.List;

import org.springframework.data.domain.Page;

import com.project.crowdfund.model.Funds;

public interface FundsService {

    Funds saveFunds(Funds funds);

    Page<Funds> getAllFunds(Integer pageNo, Integer pageSize);

    List<Funds> getStudentsByFunder(String email);

}

File: fund-frontier/fundfrontier/src/app/services/funds.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from 'src/environments/environment';
import { Funds } from '../model/funds';
import { Page } from '../model/page';
import { Observable } from 'rxjs/internal/Observable';

@Injectable({
  providedIn: 'root'
})
export class FundsService {

  fundUrl = environment.fundsBaseUrl
  constructor(private http: HttpClient) { }

  getAllFunds(pageIndex: number, pageSize: number): Observable<Page<Funds>> {
    const param = new HttpParams()
      .set('pageIndex', pageIndex)
      .set('pageSize', pageSize);
    return this.http.get<Page<Funds>>(`${this.fundUrl}/getall`, { params: param });
  }

  saveFund(funds: Funds): Observable<Funds> {
    return this.http.post<Funds>(`${this.fundUrl}/save`, funds);
  }

  getStudentsByFunder(email: string): Observable<Funds[]> {
    return this.http.get<Funds[]>(`${this.fundUrl}/findFunder/${email}`);
  }

}
